cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/handler/async/runnable/RunnableUtils.java

```diff
@@ -93,7 +93,11 @@ public static void populateRackInfoForReplicationFactorChange(Map<Short, Set<Str
     }
 
     topicsByReplicationFactor.forEach((replicationFactor, topics) -> {
-      if (replicationFactor > brokersByRack.size()) {
+      if (replicationFactor > rackByBroker.size()) {
+        throw new RuntimeException(String.format("Unable to change replication factor (RF) of topics %s to %d since there are only %d "
```

Comment:
I'm not sure Set<String>.toString() will always convert all the topics toString, it's behavior is not stable between different versions of Java.

Reply:
I can't recall observing issues related to `toString()` call on Sets using `openjdk-jdk8`, `openjdk-jdk11`, or `zulu 8`. That being said, our repo has been using ootb set implementations , such as `HashSet`, `TreeSet`, and `LinkedHashSet` -- and openJDK implementation for their `toString()` implementation in `AbstractCollection.java` hasn't changed from 8 to 11.

Have you observed issues in specific implementations / versions of Java or specific implementation of collections?

Reply:
Never mind.  It's some trauma I experienced in some previous version of Java.  This program never truncates; even for very large numbers and different Set implementations.
```java
import java.util.*;

class tostring {
   public static void main(String[] argv) throws Exception {
        Set<Integer> s = new TreeSet<>();
        int count = Integer.parseInt(argv[0]);
        for (int i =0; i < count; i++) {
           s.add(i);
        }
        
        System.out.println(s);
   }
}
```